src/App.tsx

```diff
@@ -1,24 +1,228 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import { FC, useEffect, useState } from 'react';
+import {
+  postTodo,
+  getTodos,
+  patchTodo,
+  removeTodo,
+} from './api/todos';
+import { Notification } from './components/Notification';
+import { ErrorMessage } from './types/ErrorMessage';
+import { TodoList, getVisibleTodos } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { FilteringMethod, Footer } from './components/Footer';
+import './App.scss';
 
-const USER_ID = 0;
+const USER_ID = 6648;
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isInputDisabled, disableInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState<FilteringMethod>('All');
+  const [loadingTodosIDs, setLoadingTodosIDs] = useState<number[]>([]); // problem is here
+  const [textFieldValue, setTextFieldValue] = useState('');
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.NONE);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(result => {
+        setTodos(result);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.LOAD);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.NONE);
+        }, 3000);
+      });
+  }, []);
+
+  const addTodo = () => {
+    disableInput(true);
+
+    const newTodo = {
+      title: textFieldValue,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      ...newTodo,
+      id: 0,
+    });
+
+    postTodo(newTodo)
+      .then(result => {
+        setTodos(prev => [
+          ...prev,
+          result,
+        ]);
+
+        setTextFieldValue('');
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.ADD);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.NONE);
+        }, 3000);
+      })
+      .finally(() => {
+        disableInput(false);
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodo = (id: number) => {
+    setLoadingTodosIDs(prevState => [...prevState, id]);
+
+    return removeTodo(id)
+      .then(() => {
+        const newTodosList = todos.filter(todo => todo.id !== id);
+
+        setTodos(newTodosList);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.DELETE);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.NONE);
+        }, 3000);
```

Comment:
you repeat this logic alot, let's create callback that will accept needed ErrorMessage and call it in every catch